Source/Core/InputCommon/ControllerInterface/SDL/SDL.cpp

```diff
@@ -33,39 +40,142 @@ static std::string GetJoystickName(int index)
 #endif
 }
 
-void Init()
+static void OpenAndAddDevice(int index)
 {
-#ifdef USE_SDL_HAPTIC
-  if (SDL_Init(SDL_INIT_JOYSTICK | SDL_INIT_HAPTIC) >= 0)
+  SDL_Joystick* dev = SDL_JoystickOpen(index);
+  if (dev)
   {
-    // Correctly initialized
+    auto js = std::make_shared<Joystick>(dev, index);
+    // only add if it has some inputs/outputs
+    if (js->Inputs().size() || js->Outputs().size())
```

Comment:
AFAIK, `!empty()` is preferred for such checks

Reply:
Done